src/lib/statistics/statistics_objects/range_filter.cpp

```diff
@@ -270,43 +255,68 @@ bool RangeFilter<T>::does_not_contain(const PredicateCondition predicate_conditi
 
-      Assert(variant_value2, "Between operator needs two values.");
+      Assert(variant_value2, "Between predicate needs two values.");
       const auto value2 = boost::get<T>(*variant_value2);
+      const auto is_lower_inclusive = is_lower_inclusive_between(predicate_condition);
+      const auto is_upper_inclusive = is_upper_inclusive_between(predicate_condition);
 
-      // a BETWEEN 5 AND 4 will always be empty
-      if (value2 < value) {
+      // "X BETWEEN Y AND Z" is equivalent to "X>=Y AND X<=Z" [SQL-92]. Thus, `a BETWEEN 5 AND 4` will always be empty.
+      const auto is_inclusive = is_lower_inclusive && is_lower_inclusive;
+      if (is_inclusive ? value2 < value : value2 <= value) {
         return true;
       }
 
-      // Smaller than the segment's minimum.
-      if (does_not_contain(PredicateCondition::LessThanEquals, std::max(value, value2))) {
+
+      // For the following code, consider the running example of a RangeFilter with two ranges: [2, 4] [7, 10].
+      // The predicate is `a BETWEEN x AND y`.
+      // Case (i): y is less than the segment's minimum. If y < 2 (or <= 2 for upper exclusive predicate), the predicate
+      //           cannot match.
+      const auto min = ranges.front().first;
+      const auto is_below_min = is_upper_inclusive_between(predicate_condition) ? (value2 < min) : (value2 <= min);
+      if (is_below_min) {
         return true;
       }
 
-      // Larger than the segment's maximum.
-      if (does_not_contain(PredicateCondition::GreaterThanEquals, std::min(value, value2))) {
+      // Case (ii): x is greater than than the segment's maximum. If x > 10 (or >= 10 for lower exclusive predicate),
+      //            the predicate cannot match.
+      const auto max = ranges.back().second;
+      const auto is_above_max = is_lower_inclusive_between(predicate_condition) ? (value > max) : (value >= max);
+      if (is_above_max) {
         return true;
       }
 
-      const auto range_comp = [](const std::pair<T, T> range, const T compare_value) {
-        return range.second < compare_value;
-      };
-      // Get value range or next larger value range if searched value is in a gap.
-      const auto start_lower = std::lower_bound(ranges.cbegin(), ranges.cend(), value, range_comp);
-      const auto end_lower = std::lower_bound(ranges.cbegin(), ranges.cend(), value2, range_comp);
-
-      const bool start_in_value_range =
-          (start_lower != ranges.cend()) && (*start_lower).first <= value && value <= (*start_lower).second;
-      const bool end_in_value_range =
-          (end_lower != ranges.cend()) && (*end_lower).first <= value2 && value2 <= (*end_lower).second;
+      // Case (iii): The predicate can match the segment's values. However, the predicate values can be exactly in a gap
+      //             between two ranges (e.g., a BETWEEN 5 AND 6).
+      //             We know that y >(=) 2. Thus, we find the range that starts at an element >= y. If we cannot find
```

Comment:
I always stumble over "an element" here. Maybe "find the range whose minimum is >= y" ?